Third-party-libraries.md

```diff
@@ -0,0 +1,145 @@
+
+#### Protobuf
+
+We also install the protoc and buf tools for protobuf (these versions are pinned). By the time we install those tools, we've already called `install_third_party.main()`, so we have downloaded the proto files. We use buf to generate code from those proto files (i.e. compile the proto files).
+
+### `install_third_party.py`
+
+#### Manifest files
+
+We use a `manifest.json` file to specify other dependencies that we have but which aren't Python packages or Node.js modules. This file has a `dependencies` key, under which it has the following keys:
+
+* `proto`: Files here are installed to `third_party/`
+* `frontend`: Files here are installed to `third_party/static/`
+* `oppiaTools`: Files here are installed to `oppia_tools/`
+
+Under each of these keys is a collection of key-value pairs where each key is a dependency name and each value is an object describing how to download and install the dependency. We support three types of dependencies:
+
+* Zip archives of files. The zip archive must expand to a single file or folder. The following keys are required in the object describing this dependency:
+
+  * `version`: The dependency's version string.
+  * `url`: The full URL to the zip archive.
+  * `downloadFormat`: Must be set to `zip`.
+
+  The object must specify one of the following keys:
+
+  * `rootDir`: The base name of the expanded file or folder.
+  * `rootDirPrefix`: The prefix of the base name of the expanded file or folder. The prefix concatenated with the version string should produce the full base name.
+
+  The object must also specify one of the following keys:
+
+  * `targetDir`: The base name to which the expanded file or folder should be renamed.
+  * `targetDirPrefix`: The prefix of the base name to which the expanded file or folder should be renamed. The prefix concatenated with the version string should produce the full base name.
+
+* Collections of files. The following keys are required in the object describing this dependency:
+
+  * `version`: The dependency's version string.
+  * `url`: The prefix of the URL to the files.
+  * `downloadFormat`: Must be set to `files`.
+  * `files`: List of the files to download. Each file will be downloaded from the URL formed by joining `url` with the item in `files`, using a slash (`/`) as a delimiter. Files are not renamed after they are downloaded.
+  * `targetDirPrefix`: A string which, when suffixed with `version`, yields the base name of the folder to which each file will be downloaded.
+
+* Tar archives of files. These work just like zip archives, except that there are no optional keys. Instead of the `rootDir`, `rootDirPrefix`, `targetDir`, and `targetDirPrefix` keys, these two keys are required:
+
+  * `tarRootDirPrefix`: Same as `rootDirPrefix` for zip files.
+  * `targetDirPrefix`: Same as `targetDirPrefix` for zip files.
+
+New dependencies should not be added to `manifest.json`, as this method of installing dependencies is deprecated.
```

Comment:
Perhaps explain what replaces it, and the procedure the developers should follow when adding a new third-party lib (which includes getting approval from @vojtechjelinek and me, I think).

Reply:
I've added an explanation of what replaces it. I'll include the approval process in the section on how to add third-party libs